### DISCUSSION TITLE
How to use UNION/UNION_ALL?

### DISCUSSION THREAD

**Discussion by darylhjd** [go-jet/jet#301] [Q&A]
Following this issue https://github.com/go-jet/jet/issues/69, I am trying to add a variable number of expressions into UNION.

```golang
	var firstArg SelectStatement
	tuples := make([]SelectStatement, 0, len(rows)-1)
	for idx, row := range rows {
		statement := SELECT(
			String(row.UserID),
			... // more columns
		)

		if idx == 0 {
			firstArg = statement
		} else {
			tuples = append(tuples, statement)
		}
	}

	inputRows := CTE(
		"input_rows",
	).AS(
		UNION(firstArg, firstArg, tuples...),
	)
```

Since the function requires at least 2 arguments, I am duplicating the first row, and then the rest of the input is programmatically created.

Problem with this approach is that the type of the slice is wrong, and `jet.SerializableStatement` doesn't seem to be an exportable type by the package, so I am stuck. What is the correct way to do this?

Maybe a more general question: Given a slice of structs, I would like to convert it into a table that I can use. How do I do it?

**Comment by houtn11** [comment]
Maybe something like this:

```golang
	// first two rows       
	stmt := UNION(
		SELECT(String(row[0].UserID).AS("first")/*, more columns*/),
		SELECT(String(row[1].UserID)/*, more columns*/),
	)

	// the rest of rows 
	for i := 2; i < len(rows); i++ {
		stmt = UNION(
			stmt,
			SELECT(String(row[i].UserID)/*, more columns*/),
		)
	}
```
